src/main/java/se/mickelus/tetra/module/data/ModuleData.java

```diff
@@ -30,6 +32,16 @@
     public ResourceLocation[] improvements = new ResourceLocation[0];
     public ModuleVariantData[] variants = new ModuleVariantData[0];
 
+    /**
+     * Custom data for use by addon mods. Consult the addon mod for details on their own format.
+     *
+     * Json format:
+     * {
+     *     "example_mod:custom_data": (format specified by addon mod)
+     * }
+     */
+    public Map<ResourceLocation, Object> custom = new HashMap<>();
```

Comment:
The resource locations are to encourage the use of namespaces to avoid conflicts, where different add-ons may expect different types for the same field.

In the absence of a type to deserialize to (the type parameter is just `Object` of course), Gson just deserializes to maps, arrays or the relevant primitives, which should be workable.

Though I hadn't tested, I doubt adding extra deserializers would even work, since Gson has no way of knowing what to deserialize to anyway.